### PR TITLE
feat: add notebooks to universal search

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

Notebooks do not appear in any universal search right now

## Changes

Adds notebooks to universal search.
- Only real big change here is moving the search key from `s` to `search` because that is what the universal search is setup to rely on by default

## How did you test this code?

By 👁️ 
search

### CODE REVIEW

File: posthog/api/notebook.py [hunk 1]
```diff
@@ -250,8 +250,8 @@ def _filter_request(self, request: request.Request, queryset: QuerySet) -> Query
                 queryset = queryset.filter(
                     last_modified_at__lt=relative_date_parse(request.GET["date_to"], self.team.timezone_info)
                 )
-            elif key == "s":
-                queryset = queryset.filter(title__icontains=request.GET["s"])
+            elif key == "search":
```
Comment (pauldambra) [inline posthog/api/notebook.py:253]: I'll be very disappointed in myself if this doesn't cause any backend tests to fail 🤣

**Review by pauldambra — APPROVED** [review]
🙌

**Comment by benjackwhite** [comment]
Next step - Notebooks embedded within Notebooks? 🤔

**Review by pauldambra — COMMENTED** [review]
I was going to merge this and then wondered.... does this respect

File: frontend/src/lib/components/TaxonomicFilter/taxonomicFilterLogic.tsx [hunk 2]
```diff
@@ -419,6 +420,16 @@ export const taxonomicFilterLogic = kea<taxonomicFilterLogicType>({
                         render: InlineHogQLEditor,
                         getPopoverHeader: () => 'HogQL',
                     },
+                    {
```
[the remark this answers is not in the document]
Comment (daibhin) [reply frontend/src/lib/components/TaxonomicFilter/taxonomicFilterLogic.tsx:423]: Yes, genius! I knew I was forgetting something. Will improve it

**Pull request merged**